src/main/scala/scorex/core/network/NetworkController.scala

```diff
@@ -60,15 +60,8 @@ class NetworkController(settings: NetworkSettings,
             val extAddr = intfAddr.getAddress
             myAddrs.contains(extAddr)
           }
-        } match {
-          case true => true
-          case false =>
-            if (settings.upnpEnabled) {
-              val extAddr = upnp.externalAddress
-              myAddrs.contains(extAddr)
-            } else false
-        }
-      }.recover { case t: Throwable =>
+        } || (settings.upnpEnabled && myAddrs.exists(_ == upnp.externalAddress))
```

Comment:
I should check the Wart, I remember that the complain was related to the expression `myAddrs.contains(extAddr)`
The `contains` method tends to be problematic.
The solution was to simply change the `contains` for the use of `exists`. 
The simplification made was not suggested, I just changed it because the previous code was really not elegant to me.